### ISSUE TITLE
better variational distribution for library size

### ISSUE THREAD

**Issue by jeff-regier** [YosefLab/scVI#73]

**Comment by jeff-regier** [comment]
Probably we also want a "prior" on library size that depends on batch id, aka a conditional prior. I'm not sure whether we already have that. In the paper, library size doesn't depend on batch id but it probably should -- especially if one batch is scRNA-seq data and the other is smFISH data.

**Comment by maxime1310** [comment]
Currently the framework for dealing with multiple batches would be to instanciate the dataset using the "get_attributes_from_list" method (and giving it a gene-expression matrix for each batch), that yields a batch-specific prior.  I agree that it is particularly important when dealing with different technologies that might have very different library size distributions.

**Comment by maxime1310** [comment]
@jeff-regier, I implemented what we discussed to try to improve the variational distribution (namely dropping the log-nomal for a standard normal), but it gives poorer and more unstable scores for log-likelihood (probably because there's not the KL term anymore to roughly guide the encoder).
To be sure that it was worth it to get a better variational distribution, I ran scVI without trying to learn the library size (takin the exact value everytime rather than encoding/decoding it).
It converges faster but overall does not give a better likelihood than the one in Romain's paper.

**Comment by jeff-regier** [comment]
Isn't there still a KL term for library size? It's just a KL divergence btw a normal  (the variational dist) and a standard normal (the prior), rather a KL divergence between a log normal and a log normal, right?

**Comment by maxime1310** [comment]
Yes there is still a KL term, what I wanted to say was that in the case of the log-normal prior the KL term actually gives the model information on the mean and variance of the log library size, while the KL between the normal and standard normal doesn't.

**Comment by jeff-regier** [comment]
OK, let's make sure we're calculating log p(x) correctly, as discussed. If it still isn't better, we can probably conclude that a log normal isn't that bad of a variational distribution for library size.

**Comment by jeff-regier** [comment]
@maxime1310 should we close this and stick with the variational distribution we have? It seems like log normal may be pretty good after all.
